src/devices/Mcp3428/Mcp342x.cs

```diff
@@ -0,0 +1,387 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device.I2c;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Mcp3428
+{
+    public abstract class Mcp342x : IDisposable
+    {
+        /// <summary>
+        /// Protected constructor to easily generate MCP3426/7 devices whose only difference is channel count and I2C address
+        /// </summary>
+        /// <param name="i2CDevice">The i2 c device.</param>
+        /// <param name="channels">The channels.</param>
+        /// <autogeneratedoc />
+        protected Mcp342x(I2cDevice i2CDevice, int channels)
+        {
+            _i2cDevice = i2CDevice;
+            ChannelCount = channels;
+            ReadValue(); // Don't like this in constructor, makes sure props are valid
+        }
+
+        /// <summary>
+        /// Gets the last transmitted bytes. Debug function
+        /// </summary>
+        /// <value>The last bytes.</value>
+        /// <autogeneratedoc />
+        public ReadOnlyMemory<byte> LastBytes => _readBuffer;
+
+        /// <summary>
+        /// Channel most recently read
+        /// </summary>
+        /// <value>The last channel.</value>
+        /// <autogeneratedoc />
+        public byte LastChannel => LastConversion.Channel;
+
+        public ModeEnum Mode
+        {
+            get => _mode;
+            set
+            {
+                WriteConfig(Helpers.SetModeBit(LastConfigByte, value));
+                _mode = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the input gain.
+        /// </summary>
+        /// <value>The pga gain.</value>
+        /// <autogeneratedoc />
+        public GainEnum PGAGain
```

Comment:
if I read correctly PGA stands for `Programmable-gain amplifier` which means this name reads weird: `Programmable gain amplifier gain`

Reply:
It's called this in the device documentation.

Reply:
we can fix it here but treat as optional if that's what the doc calls it :)

Reply:
Then I will name it InputGain.